src/components/ShareButton/ShareButtonContainer.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC } from 'react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { QUERY_KEYS } from 'api';
+import { notesActions } from 'api/queries';
+import { useAppSelector } from 'store/hooks';
+import { getUser } from 'store/selectors';
+import { INote } from 'types';
+
+import { StyledShareIcon } from './styles';
+import { IShareButtonContainer } from './types';
+
+const ShareButtonContainer: FC<IShareButtonContainer> = ({ note }) => {
+  const { user } = useAppSelector(getUser);
+  const isNoteEditable = user?.userId === note.userId;
+
+  const editSharedStatus = (note: INote): INote => {
+    return { ...note, isShared: !note.isShared };
```

Comment:
isShared: !note.isShared 
is it ok?

Reply:
I need to toggle the boolean value of isShared status. I can do this using the logical NOT operator or the ternary operator . I thought that it will be simpler to use ! instead of ? : false : true.